crates/aptos-ledger/src/lib.rs

```diff
@@ -0,0 +1,260 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use hex::encode;
+use ledger_apdu::APDUCommand;
+use ledger_transport_hid::{hidapi::HidApi, TransportNativeHID};
+use std::str;
+
+const DERIVATIVE_PATH: &str = "m/44'/637'/0'/0'/0'"; // TODO: Add support for multiple index
+
+const CLA_APTOS: u8 = 0x5B; // Aptos CLA Instruction class
+const INS_GET_VERSION: u8 = 0x03; // Get version instruction code
+const INS_GET_APP_NAME: u8 = 0x04; // Get app name instruction code
+const INS_GET_PUB_KEY: u8 = 0x05; // Get public key instruction code
+const INS_SIGN_TXN: u8 = 0x06; // Sign the transaction
+const APDU_CODE_SUCCESS: u16 = 36864; // success code for transport.exchange
+
+const MAX_APDU_LEN: usize = 255;
+const P1_NON_CONFIRM: u8 = 0x00;
+const P1_CONFIRM: u8 = 0x01;
+const P1_START: u8 = 0x00;
+const P2_MORE: u8 = 0x80;
+const P2_LAST: u8 = 0x00;
+
+#[derive(Debug)]
+pub enum AptosLedgerError {
+    /// Error when trying to open a connection to the Ledger device
+    DeviceNotFound,
+    /// Unexpected error
+    UnexpectedError(String),
+}
+
+/// Returns the current version of the Aptos app on Ledger
+pub fn get_app_version() -> Result<String, AptosLedgerError> {
+    // open connection to ledger
+    let transport = match open_ledger_transport() {
+        Ok(transport) => transport,
+        Err(err) => return Err(err),
+    };
+
+    match transport.exchange(&APDUCommand {
+        cla: CLA_APTOS,
+        ins: INS_GET_VERSION,
+        p1: P1_NON_CONFIRM,
+        p2: P2_LAST,
+        data: vec![],
+    }) {
+        Ok(response) => {
+            // Received response from Ledger
+            if response.retcode() == APDU_CODE_SUCCESS {
+                let major = response.data()[0];
+                let minor = response.data()[1];
+                let patch = response.data()[2];
+                let version = format!("{}.{}.{}", major, minor, patch);
```

Comment:
Instead of turning this into a string, could we make a `Version` struct with major, minor, and patch fields, and then impl `Display` for it?

Reply:
Good call!